loadtests/__init__.py

```diff
@@ -1,54 +1,169 @@
+import json
+import random
+import uuid
+
 from requests.auth import HTTPBasicAuth
 from loads.case import TestCase
 
 
-class TestPOC(TestCase):
+PERCENTAGE_OF_CREATION = 20
+PERCENTAGE_OF_UPDATE = 50
+PERCENTAGE_OF_FILTERING = 60
+PERCENTAGE_OF_READ_POSITION = 80
+PERCENTAGE_OF_MARKING_AS_READ = 40
+PERCENTAGE_OF_ARCHIVING = 10
+PERCENTAGE_OF_DELETING = 10
+PERCENTAGE_OF_CONFLICT_CREATION = 10
+PERCENTAGE_OF_CONFLICT_UPDATE = 10
+
+
+class TestBasic(TestCase):
+    available_users = [
+        ('alice', 'secret'),
+        ('bob', 'secret'),
+        ('charlie', 'secret')
+    ]
+
     def __init__(self, *args, **kwargs):
-        super(TestPOC, self).__init__(*args, **kwargs)
+        super(TestBasic, self).__init__(*args, **kwargs)
+
+        self.random_user = None
 
-        self.alice_auth = HTTPBasicAuth('alice', 'secret')
-        self.john_auth = HTTPBasicAuth('john', 'secret')
+        # Create at least one record per user
+        for user in self.available_users:
+            self.basic_auth = HTTPBasicAuth(*user)
+            self.test_create_record()
+
+    def setUp(self):
+        # Change current user randomly
+        self.random_user = random.choice(self.available_users)
+        self.basic_auth = HTTPBasicAuth(*self.random_user)
+
+        resp = self.session.get(self.api_url('articles'), auth=self.basic_auth)
+        self.random_record = random.choice(resp.json()['items'])
+        self.random_id = self.random_record['_id']
+        self.random_url = self.api_url('articles/{0}'.format(self.random_id))
+
+    def incr_counter(self, name):
+        if not self.random_user:
+            return
+        name = "{} {}".format(self.random_user[0], name)
+        super(TestBasic, self).incr_counter(name)
 
     def api_url(self, path):
-        return "{0}/v1/{1}".format(self.server_url, path)
+        return "{0}/v0/{1}".format(self.server_url, path)
 
     def test_all(self):
-        self.test_home()
-        self.test_create_record()
-        self.test_update_status()
-        self.test_list_filtered()
+        if random.randint(0, 100) < PERCENTAGE_OF_CREATION:
```

Comment:
I would create an intermediate method to check if we should do this test or not. I probably also would rework all these constants in a python dict and have the key be the subject, so we could do something like `if self.should_run('creation'):`